hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/ClientGSIContext.java

```diff
@@ -77,12 +82,35 @@ public void updateResponseState(RpcResponseHeaderProto.Builder header) {
   @Override
   public synchronized void receiveResponseState(RpcResponseHeaderProto header) {
     if (header.hasRouterFederatedState()) {
-      routerFederatedState = header.getRouterFederatedState();
+      routerFederatedState = mergeRouterFederatedState(header.getRouterFederatedState());
+      routerFederatedState = mergeRouterFederatedState(header.getRouterFederatedState());
     } else {
       lastSeenStateId.accumulate(header.getStateId());
     }
   }
 
+  /**
+   * Merge the local FederatedState and RemoteFederateState to get the max value for each namespace.
+   * @param remoteFederateState the remote RouterFederatedState.
+   * @return one ByteString object which contains the max value of each namespace.
+   */
+  private ByteString mergeRouterFederatedState(ByteString remoteFederateState) {
+    Map<String, Long> localMapping = new HashMap<>(
+        ByteStringUtil.getRouterFederatedStateMap(this.routerFederatedState));
+    Map<String, Long> remoteMapping =
+        ByteStringUtil.getRouterFederatedStateMap(remoteFederateState);
+
+    for (String ns : remoteMapping.keySet()) {
+      long remoteValue = remoteMapping.get(ns);
+      long localValue = localMapping.getOrDefault(ns, 0L);
+      localMapping.put(ns, Math.max(remoteValue, localValue));
+    }
+    HdfsServerFederationProtos.RouterFederatedStateProto.Builder federatedStateBuilder =
```

Comment:
To make it shorter to read, can you import org.apache.hadoop.hdfs.federation.protocol.proto.HdfsServerFederationProtos.RouterFederatedStateProto?